Sources/Kitura/CodableRouter.swift

```diff
@@ -600,21 +603,38 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    /// Precondition: Path is known to contain a : character
+    func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
+        let paramaterString = path.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
+        let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
+        return parameter.isEmpty || parameter.elementsEqual("id")
+    }
+
+    func pathSyntaxIsValid(_ path: String, identifierExpected: Bool) -> Bool {
+        let identifierSupplied = path.contains(":")
+        if !identifierExpected && identifierSupplied {
+            Log.error("Erroneous path '\(path)' is not allowed. Codable routes do not allow parameters.")
+            return false
+        }
+        if identifierExpected && !identifierSupplied {
+            Log.warning("Identifier expected for the '\(path)'. Id parameter will be automatically appeneded by default to the path .")
```

Comment:
`Identifier expected for '\(path)'.` (remove 'the')
`appeneded` -> `appended`
We don't really need "`by default`".  The last statement could simply be: `:id will be automatically appended to the path.`